tests/tests/test-crowdloan.ts

```diff
@@ -0,0 +1,283 @@
+  });
+});
+describeDevMoonbeam(
+  "should be able to register the genesis account for reward - with small amount",
+  (context) => {
+    let genesisAccount: KeyringPair;
+    let alithAccount: KeyringPair;
+    const relayChainAddress: string = "couldBeAnyString";
+    before("Setup genesis account for substrate", async () => {
+      const keyring = new Keyring({ type: "ethereum" });
+      genesisAccount = await keyring.addFromUri(GENESIS_ACCOUNT_PRIVATE_KEY, null, "ethereum");
+      alithAccount = await keyring.addFromUri(ALITH_PRIV_KEY, null, "ethereum");
+    });
+
+    it("initializeRewardVec", async function () {
+      await context.polkadotApi.tx.sudo
+        .sudo(
+          context.polkadotApi.tx.crowdloanRewards.initializeRewardVec(
+            [[relayChainAddress, ALITH, 3_000_000n * GLMR]],
+            0,
+            1
+          )
+        )
+        .signAndSend(genesisAccount);
+      await context.createBlock();
+      expect(
+        ((await context.polkadotApi.query.crowdloanRewards.accountsPayable(ALITH)).toHuman() as any)
+          .total_reward
+      ).to.equal("3.0000 MUnit");
+    });
+
+    it("showMeTheMoney", async function () {
+      await context.polkadotApi.tx.crowdloanRewards.showMeTheMoney().signAndSend(alithAccount);
+      await context.createBlock();
+      const isPayable4 = await context.polkadotApi.query.crowdloanRewards.accountsPayable(ALITH);
+      expect((isPayable4.toHuman() as any).claimed_reward).to.equal("600.0059 kUnit");
+    });
+  }
+);
+describeDevMoonbeam("should be able to register many accounts", (context) => {
+  let genesisAccount: KeyringPair;
+  const relayChainAddress: string = "couldBeAnyString";
+  const numberOfAccounts: number = 5000; // min 2
+  let largInput: [string, string, bigint][];
+
+  before("Setup genesis account for substrate", async () => {
+    const keyring = new Keyring({ type: "ethereum" });
+    genesisAccount = await keyring.addFromUri(GENESIS_ACCOUNT_PRIVATE_KEY, null, "ethereum");
+  });
+
+  it("should create a bunch of test eth accounts", async function () {
+    this.timeout(20000);
+    let web3 = new Web3();
+    let accounts = new Array(numberOfAccounts).fill(0).map((_, i) => web3.eth.accounts.create());
+    largInput = accounts.map((acc: Account, i: number) => {
+      return [relayChainAddress + i.toString(), acc.address, 1000n * GLMR];
+    });
+    expect(largInput.length).to.eq(numberOfAccounts);
+    expect(largInput[0][1] !== largInput[numberOfAccounts - 1][1]).to.eq(true);
+  });
+  it("should be able to register many accounts : " + numberOfAccounts, async function () {
+    this.timeout(20000);
+    await context.polkadotApi.tx.sudo
+      .sudo(context.polkadotApi.tx.crowdloanRewards.initializeRewardVec(largInput, 0, largInput))
+      .signAndSend(genesisAccount);
+    await context.createBlock();
+    await Promise.all(
+      largInput.map((input) => {
+        return new Promise<void>(async (res, reject) => {
```

Comment:
No need to create promise here, the code executed is already a promise

Reply:
expect returns a Promise?

Reply:
ha yes indeed